### PR TITLE
pscanrules: Adjust Mac OSX Salted SHA-1 name, risk, and confidence

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Add change note.
- HashDisclosureScanRule > Adjust pattern's name, risk, and confidence.
- HashDisclosureScanRuleUnitTest > Adjust assertions related to this pattern/alert.
- pscanrules.html > Added help note outlining the confidence/threshold association under which patterns are evaluated.

## Related Issues
Fixes zaproxy/zaproxy#8624

Other ref: https://github.com/s0md3v/Bolt/blob/37aa096db93e026f0524fe94c67dc1dae8985f3e/db/hashes.json#L291-L299

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/pscanrules/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Changed
+- Rename Mac OSX salted SHA-1 in the Hash Disclosure scan rule to "Salted SHA-1", reduce the associated alerts to Low risk and Low confidence, to align with other SHA related patterns. (Note such matches may indicate leaks related but not limited to: MacOS X, Oracle, Tiger-192, Haval-192) (Issue 8624).
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:8]: I think it's important to mention that this hash check will only run on LOW threshold now. Maybe the help too (though we didn't mention before for the other hashes).

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java [hunk 2]
```diff
@@ -103,7 +103,7 @@ void shouldRaiseAlertWhenResponseContainsCookieWithMd5Hash() throws Exception {
     }
 
     @Test
-    void shouldRaiseAlertWhenResponseContainsOsxSha1AtLowThreshold() throws Exception {
+    void shouldRaiseAlertWhenResponseContainsSaltedSha1AtLowThreshold() throws Exception {
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java:106]: This is a specific JS test, the hash type is irrelevant (should be named accordingly). `shouldNotRaiseAlertWhenResponseContainsOsxSha1InJsAtNonLowThreshold` should be updated as well.

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java [hunk 3]
```diff
@@ -112,8 +112,10 @@ void shouldRaiseAlertWhenResponseContainsOsxSha1AtLowThreshold() throws Exceptio
         scanHttpResponseReceive(msg);
         // Then
         assertThat(alertsRaised.size(), is(1));
-        assertThat(alertsRaised.get(0).getName(), is("Hash Disclosure - Mac OSX salted SHA-1"));
+        assertThat(alertsRaised.get(0).getName(), is("Hash Disclosure - Salted SHA-1"));
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java:115]: Change to `startsWith`.

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java [hunk 4]
```diff
@@ -112,8 +112,10 @@ void shouldRaiseAlertWhenResponseContainsOsxSha1AtLowThreshold() throws Exceptio
...
         assertThat(alertsRaised.get(0).getEvidence(), is(hashVal));
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(Alert.RISK_LOW)));
+        assertThat(alertsRaised.get(0).getConfidence(), is(equalTo(Alert.CONFIDENCE_LOW)));
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java:118]: This is irrelevant for the test.

**Comment by kingthorin** [comment]
Adjusted

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java [hunk 5]
```diff
@@ -112,16 +113,17 @@ void shouldRaiseAlertWhenResponseContainsOsxSha1AtLowThreshold() throws Exceptio
...
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(Alert.RISK_LOW)));
     }
 
     @ParameterizedTest
     @EnumSource(
             value = Plugin.AlertThreshold.class,
             names = {"HIGH", "MEDIUM"})
-    void shouldNotRaiseAlertWhenResponseContainsOsxSha1InJsAtNonLowThreshold(
-            AlertThreshold threshold) throws Exception {
+    void shouldNotRaiseAlertWhenResponseContainsHashInJsAtNonLowThreshold(AlertThreshold threshold)
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/HashDisclosureScanRuleUnitTest.java:125]: e.g. `shouldNotRaiseAlertWhenJavascriptResponseContainsHashAtNonLowThreshold` so the two methods have consistent naming.

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 6]
```diff
@@ -228,9 +228,14 @@ <H2 id="id-10033">Directory Browsing</H2>
...
 Passively scans for password hashes disclosed by the web server. <br>
-Various formats are including, including some formats such as MD4, MD5, and SHA*, which are sometimes used for purposes other than to contain password hashes.
+Various formats are included, including some formats such as MD4, MD5, and SHA*, which are sometimes used for purposes other than to contain password hashes.
 <br>
-<strong>Note:</strong> This scan rule will only analyze text responses, and only analyze JavaScript responses on LOW Threshold. 
+<strong>Note:</strong>
+<ul>
+    <li>This scan rule will only analyze text responses, and only analyze JavaScript responses on LOW Threshold.<li>
+    <li>The selection of Hash patterns which are evaluated is tied to the Confidence we have in the pattern and the Threshold set for the scan rule.
+        In other words: Low confidence patterns will only be included for evaludation at Low Threshold, etc.</li>
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:237]: Typo in evaluation.

**Comment by kingthorin** [comment]
Got those

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**